ksql-engine/src/main/java/io/confluent/ksql/topic/TopicDeleteInjector.java

```diff
@@ -75,6 +78,26 @@ public TopicDeleteInjector(
         Objects.requireNonNull(schemaRegistryClient, "schemaRegistryClient");
   }
 
+  private void checkTopicRefs(final DataSource<?> source) {
+    final String topicName = source.getKafkaTopicName();
+    final String sourceName = source.getName();
+    final Map<String, DataSource<?>> sources = metastore.getAllDataSources();
+    final List<String> using = sources.values().stream()
+        .filter(s -> s.getKafkaTopicName().equals(topicName))
+        .map(s -> s.getName())
+        .filter(name -> !sourceName.equals(name))
+        .collect(Collectors.toList());
```

Comment:
Rather than create a list, you can just build the string you need later:

```
final String using = sources.values().stream()
        .filter(s -> s.getKafkaTopicName().equals(topicName))
        .map(s -> s.getName())
        .filter(name -> !sourceName.equals(name))
        .collect(Collectors.joining(", "));

if (!using.isEmpty()) {
      throw new KsqlException(
          String.format(
              "Refusing to delete topic. Found other data sources (%s) using topic %s",
             using,
              topicName
          )
      );
    }
```